Update Project-5-Task-1.md

JaronJCheung opened the pull request:
Dear @dmcnulty27, @emmoramor, @Jiemingy, @katiejohnso, @jvonderhbuff, @lchill2, @cameronmore, @jlbillig,

Greetings Group 2! Delaney, brought to my attention that Professor Beverley wanted us to review each other's group projects. As such, I have tagged you all to review our work for Project 5, task 1. 

We have not completed our work yet and we would kindly like to ask if you could review our work on Sunday, when we will have completed the majority of the work. I will send you all a notification when we are ready for a review, if that's okay :). 

Thanks in advance for your patience and consideration. 

My repository will be the primary repository for Group 1's work on task 1 of Project 5. Group 1 is constituted by @peihongx, @ahasanzadeh13, @giacomodecolle, @Arachnophobian, @Finn1928, @oliviahobai, @tmprd, and myself. 

Excited to learn from you all and exchange learnings! 

Peace and luv, 
Jaron

JaronJCheung committed:
Update Project-5-Task-1.md

tmprd committed:
Sample SHACL shape + instance

tmprd committed:
Update BFO-SHACL-Instances.md

tmprd committed:
Update BFO-SHACL-Instances.md

JaronJCheung committed:
BFO 2020 Knowledge Graph with Instance Data Added

JaronJCheung committed:
BFO SHACL Schema 3.0 for BFO 2020 Knowledge Graph with Instance Data

JaronJCheung committed:
Update Project-5-Task-1.md

tmprd committed:
Merge branch 'JaronJCheung-P-5' into Tim-Project-5

tmprd committed:
Add some of my sense data

tmprd committed:
Try to fix weird spacing change

tmprd committed:
Fix IRI

tmprd committed:
Make some contradictions + add Berkley's mind

tmprd committed:
Update validations doc

tmprd committed:
Update instances + validation result

tmprd committed:
Update bad proposition

dmcnulty27 commented:
Methodology Review
Group 1 first of all, wow. You have done a lot of work and it is quite impressive. Your axiom knowledge bases are incredibly expansive. Group 2 would like to commend you on your effort and collaboration! I would suggest that you scale down a little bit 😊 but I think you already know that! May I ask how you plan to check your schemas? I am interested to see how you will check the schemes you present. Further, it would be beneficial to see how and why you choose the axioms you did and how this will fit into your schema constructions. I would love to hear about the knowledge graph. I know y’all are working on developing the graph, but I would personally benefit from hearing up on how y’all plan to design the graph. Additionally, will you build one graph or several? 
If there is more you would like us to consider please let me know! Overall, Great Work!!

dmcnulty27 commented on Project-5-Task-1.md:
```diff
@@ -1302,9 +1308,7 @@ Zero dimensional spatial region, one dimensional spatial region, two dimensional
 
 ___
```
Methodology Review
Group 1 first of all, wow. You have done a lot of work and it is quite impressive. Your axiom knowledge bases are incredibly expansive. Group 2 would like to commend you on your effort and collaboration! I would suggest that you scale down a little bit 😊 but I think you already know that! May I ask how you plan to check your schemas? I am interested to see how you will check the schemes you present. Further, it would be beneficial to see how and why you choose the axioms you did and how this will fit into your schema constructions. I would love to hear about the knowledge graph. I know y’all are working on developing the graph, but I would personally benefit from hearing up on how y’all plan to design the graph. Additionally, will you build one graph or several?
If there is more you would like us to consider please let me know! Overall, Great Work!!

JaronJCheung replied:
Dear @dmcnulty27, 

Thank you so much for your review and your encouraging words! I think we plan on scaling down for sure. I do not think we are able to finish all of these axioms 😅. 

To answer your questions:

(1) We are checking our schemas using a SHACL validator tool which you can find here: https://github.com/TopQuadrant/shacl. Also, we are using SHACL Playground as a secondary supplemental tool. 

(2) On choosing these axioms: We actually did not choose these axioms, they were assigned to us. Professor Beverley wanted us to ensure that our SHACL schema can account for the logical axioms that undergird BFO. The logical axioms have been created primarily by Alan Ruttenberg (according to my limited knowledge). 

(3) On the knowledge graph: We will be building and submitting one graph. The initial Astrea SHACL graph we created is based on the knowledge graph with instance data added, while also being built off of the BFO 2020 ontology without the first order logic axioms added. This project seeks to update that initial SHACL schema with more SHACL shapes that enforce the first order logic axioms. So all the validated SHACL shapes that you see on our Google doc that correspond to a logical axiom will be added to the original Astrea SHACL schema. 

At this link, you can see all of our most updated SHACL enforced FOL work: https://docs.google.com/document/d/1miP3WUXwh3rG3AoBgrDXQrfRwlesV-7XlMhrtjQG2sc/edit#

Thanks again for the encouragement. It's been a hard project and we appreciate the time you spent to review our work. 

Please let me know if you have any questions. 

Best, 
Jaron & Group 1

katiejohnso replied:
One more Methodology question, coming from a place of curiosity/confusion (rather than critique): There might be a question: if the ontological structure of a project which is supposed to be a BFO structure, but an entity in that project does not fit into any class in BFO, then there should be a Shacl shape checking it.

JaronJCheung replied:
@katiejohnso, thanks for sharing this curiosity! I would be very interested in that particular entity that does not fit in the BFO structure and also, how a SHACL shape would test it! Lol. I personally don't have any ideas that immediately come to mind. Your comment has me thinking now though. Thanks for the good thought!

peihongx replied:
> One more Methodology question, coming from a place of curiosity/confusion (rather than critique): There might be a question: if the ontological structure of a project which is supposed to be a BFO structure, but an entity in that project does not fit into any class in BFO, then there should be a Shacl shape checking it.

I guess any entity must fit into some class in BFO because BFO is a top-level ontology? Even if we cannot find a subclass which a strange entity falls under, we can always at least subsume it under BFO:entity.

tmprd committed:
FOL Spatial Axioms

tmprd committed:
Try another axiom...

dmcnulty27 commented:
Osiyo Group 1 
I was unable to see updates to the graphing part of your project. I am hoping this is a user error on my end. Our group is planning to meet at 1pm today to go over this work!

Jiemingy commented:
Hey, Group 1 people,

Thank you for inviting me to review your project!

The axiom knowledge you made is amazing! I think it can not only be used for your task but also for some other purposes. For your task, I suggest that you may create an example using BFO, e.g., question 5 in project 3, and then use Shacl shapes to validate it. In such a way you might get some good Shacl schemas to parallel BFO. For example, the constraint "datatype" can be used to check whether a property of a given node using BFO has a proper datatype. So by identifying each ontological entity in BFO with a proper datatype, you may create a specific Shacl schema validating datatypes of cases using BFO. 

Bests,
Jieming

tmprd committed:
Merge pull request #11 from JaronJCheung/Tim-Project-5

Tim Project 5

JaronJCheung commented:
> Osiyo Group 1
> I was unable to see updates to the graphing part of your project. I am hoping this is a user error on my end. Our group is planning to meet at 1pm today to go over this work!



> Osiyo Group 1
> I was unable to see updates to the graphing part of your project. I am hoping this is a user error on my end. Our group is planning to meet at 1pm today to go over this work!

Osiyo @dmcnulty27 and Group 2 (@Jiemingy, @katiejohnso, @Jlbillig, @Jvonderhbuff, @cameronmore, @emmoramor, @lchill2)!

Thank you again for being willing to review our project. Here are some shortcut links to each of our deliverables (so you don't have to hunt these down in the repository):

**1. BFO 2020 Knowledge Graph with instance data added:**
https://github.com/JaronJCheung/PHI-696/blob/JaronJCheung-P-5/BFO_2020_Knowledge%20Graph.ttl

This knowledge graph represents the BFO 2020 ontology. 

**2. BFO SHACL Schema File:** 
https://github.com/JaronJCheung/PHI-696/blob/JaronJCheung-P-5/BFO_SHACL_SCHEMA_3.0.ttl

This Astrea SHACL schema will have our added SHACL first order logic enforced shapes to it by tonight. 

**3. First Order Logic Enforced SHACL Shapes:**
https://docs.google.com/document/d/1miP3WUXwh3rG3AoBgrDXQrfRwlesV-7XlMhrtjQG2sc/edit#

These SHACL shapes will correspond to a first order logic axiom. They will ensure that BFO 2020 conforms to the FOL axioms that adhere to BFO standards. 

The reason we are not updating the Github repository itself is because we want to prevent the problems with having "too many cooks in the kitchen". We don't want to accidentally update and add things in an overlapping way. As such, we are doing all of our work on Google docs and then will transfer our final product to Github (today, 4/16). 

Please let me know if you have any questions. 

Thanks and take care, 
Jaron and Group 1 (@peihongx, @giacomodecolle, @ahasanzadeh13, @Arachnophobian, @Finn1928, @oliviahobai, @tmprd)

dmcnulty27 commented:
Thank you for this update Jaron!
Good luck and have fun collaborating!

JaronJCheung commented:
> Thank you for this update Jaron!
> Good luck and have fun collaborating!

Thank you, @dmcnulty27! You as well! 

Best, 
Jaron

JaronJCheung committed:
Update Project-5-Task-1.md

peihongx commented:
> > Osiyo Group 1
> > I was unable to see updates to the graphing part of your project. I am hoping this is a user error on my end. Our group is planning to meet at 1pm today to go over this work!
> 
> > Osiyo Group 1
> > I was unable to see updates to the graphing part of your project. I am hoping this is a user error on my end. Our group is planning to meet at 1pm today to go over this work!
> 
> Osiyo @dmcnulty27 and Group 2 (@Jiemingy, @katiejohnso, @Jlbillig, @Jvonderhbuff, @cameronmore, @emmoramor, @lchill2)!
> 
> Thank you again for being willing to review our project. Here are some shortcut links to each of our deliverables (so you don't have to hunt these down in the repository):
> 
> **1. BFO 2020 Knowledge Graph with instance data added:** https://github.com/JaronJCheung/PHI-696/blob/JaronJCheung-P-5/BFO_2020_Knowledge%20Graph.ttl
> 
> This knowledge graph represents the BFO 2020 ontology.
> 
> **2. BFO SHACL Schema File:** https://github.com/JaronJCheung/PHI-696/blob/JaronJCheung-P-5/BFO_SHACL_SCHEMA_3.0.ttl
> 
> This Astrea SHACL schema will have our added SHACL first order logic enforced shapes to it by tonight.
> 
> **3. First Order Logic Enforced SHACL Shapes:** https://docs.google.com/document/d/1miP3WUXwh3rG3AoBgrDXQrfRwlesV-7XlMhrtjQG2sc/edit#
> 
> These SHACL shapes will correspond to a first order logic axiom. They will ensure that BFO 2020 conforms to the FOL axioms that adhere to BFO standards.
> 
> The reason we are not updating the Github repository itself is because we want to prevent the problems with having "too many cooks in the kitchen". We don't want to accidentally update and add things in an overlapping way. As such, we are doing all of our work on Google docs and then will transfer our final product to Github (today, 4/16).
> 
> Please let me know if you have any questions.
> 
> Thanks and take care, Jaron and Group 1 (@peihongx, @giacomodecolle, @ahasanzadeh13, @Arachnophobian, @Finn1928, @oliviahobai, @tmprd)


Hi @dmcnulty27 and Group 2 (@Jiemingy, @katiejohnso, @Jlbillig, @Jvonderhbuff, @cameronmore, @emmoramor, @lchill2), our Team's substantive work are on **those shapes marked in green** in the third file (namely that Google doc file). Other shapes marked in red or yellow are still in progress. In our Google doc, you can use navigation bar on the left side to easily find our work for each section. #Born Free But Everywhere in SHACL# is a section about our workflow steps. You can skip it and then turn to Part I

As for the first and second file, the first one is a knowledge graph which can be validated by our SHACL schema, and the second one is basically a SHACL translation of the knowledge graph. Our final SHACL schema will be the second file extended by green SHACL shapes in our Google doc!

JaronJCheung committed:
Update Project-5-Task-1.md

JaronJCheung committed:
Rename BFO-SHACL-Instances.md to BFO-SHACL-Instances-Tim.md

JaronJCheung committed:
Rename BFO_2020_Knowledge Graph.ttl to 1. BFO_2020_Knowledge Graph.ttl

JaronJCheung committed:
Rename BFO_SHACL_SCHEMA_3.0.ttl to 2. BFO_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Rename Project-5-Task-1.md to 1. Project-5-Task-1.md

JaronJCheung committed:
Rename 1. BFO_2020_Knowledge Graph.ttl to 2. BFO_2020_Knowledge Graph.ttl

JaronJCheung committed:
Rename 2. BFO_SHACL_SCHEMA_3.0.ttl to 3. BFO_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Rename 2. BFO_2020_Knowledge Graph.ttl to 2. Project_5_BFO_2020_Knowledge Graph.ttl

JaronJCheung committed:
Rename 1. Project-5-Task-1.md to 1. Project_5_Task_1.md

JaronJCheung committed:
Rename 3. BFO_SHACL_SCHEMA_3.0.ttl to 3. Project_5_BFO_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

Finn1928 commented:
> Hey, Group 1 people,
> 
> Thank you for inviting me to review your project!
> 
> The axiom knowledge you made is amazing! I think it can not only be used for your task but also for some other purposes. For your task, I suggest that you may create an example using BFO, e.g., question 5 in project 3, and then use Shacl shapes to validate it. In such a way you might get some good Shacl schemas to parallel BFO. For example, the constraint "datatype" can be used to check whether a property of a given node using BFO has a proper datatype. So by identifying each ontological entity in BFO with a proper datatype, you may create a specific Shacl schema validating datatypes of cases using BFO.
> 
> Bests, Jieming

Hi! I think it would be possible to do what you're describing, though it would take us well beyond the task for this assignment. It's taken us this long to validate just half of the axioms we've been presented with. Thank you for your suggestion!

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Create Project-5

JaronJCheung committed:
Create P5

JaronJCheung committed:
Rename 1. Project_5_Task_1.md to Project-5/1. Project_5_Task_1.md

JaronJCheung committed:
Rename 2. Project_5_BFO_2020_Knowledge Graph.ttl to Project-5/2. Project_5_BFO_2020_Knowledge Graph.ttl

JaronJCheung committed:
Rename 3. Project_5_BFO_SHACL_SCHEMA_3.0.ttl to Project-5/3. Project_5_BFO_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Delete P5

JaronJCheung committed:
Delete Spatial-Axioms.ttl

JaronJCheung committed:
Delete directory directory

JaronJCheung committed:
Final BFO SHACL Schema with ALL SHACL Shapes Added (92)

JaronJCheung committed:
Rename FINAL_BFO_SHACL_Schema.ttl to 4. FINAL_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Special Collection Version BFO SHACL Schema

This is a delimited special collection of SHACL shapes from Karl, Giacomo, and Finn that returned 0 errors. However, there are violations that are returned on this knowledge graph. These violations are primarily centered on "min count 1" constraints.

JaronJCheung committed:
Rename SPECIAL_COLLECTION_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl to 5. SPECIAL_COLLECTION_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl

JaronJCheung committed:
Rename 4. FINAL_BFO_SHACL_Schema.ttl to 4. Project_5_FINAL_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Rename 5. SPECIAL_COLLECTION_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl to 5. Project_5_SPECIAL_COLLECTION_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl

JaronJCheung committed:
Rename BFO-SHACL-Instances-Tim.md to Project-5/BFO-SHACL-Instances-Tim.md

JaronJCheung committed:
Create BFO-SHACL-Instances-Tim.md

JaronJCheung committed:
Delete BFO-SHACL-Instances-Tim.md

JaronJCheung committed:
Rename 3. Project_5_BFO_SHACL_SCHEMA_3.0.ttl to 3. Project_5_TEMPLATE_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Rename 3. Project_5_TEMPLATE_SHACL_SCHEMA_3.0.ttl to 3. Project_5_TEMPLATE_BFO_SHACL_SCHEMA_3.0.ttl

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Rename 5. Project_5_SPECIAL_COLLECTION_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl to 5. Project_5_Special_Collection_BFO_SHACL_Shapes_Karl-Giacomo-Finn.ttl

JaronJCheung committed:
Delete 4. Project_5_FINAL_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Add files via upload

JaronJCheung committed:
Rename Project_5_FINAL_BFO_SHACL_Schema.ttl to 4. Project_5_FINAL_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Rename 3. Project_5_TEMPLATE_BFO_SHACL_SCHEMA_3.0.ttl to 3. Project_5_TEMPLATE_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Rename 3. Project_5_TEMPLATE_BFO_SHACL_Schema.ttl to 3. Project_5_Template_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Rename 4. Project_5_FINAL_BFO_SHACL_Schema.ttl to 4. Project_5_BFO_SHACL_Schema_FINAL.ttl

JaronJCheung committed:
Rename 4. Project_5_BFO_SHACL_Schema_FINAL.ttl to 4. Project_5_FINAL_BFO_SHACL_Schema.ttl

JaronJCheung committed:
Update 1. Project_5_Task_1.md

JaronJCheung committed:
Update 1. Project_5_Task_1.md
